src/widget/button.rs

```diff
@@ -47,34 +45,39 @@ impl Label {
         ctx.add(self, &[])
     }
 
-    fn get_layout(&self, dwrite_factory: &directwrite::Factory) -> TextLayout {
+    fn get_layout(
+        &self,
+        rt: &mut Piet,
+        font_size: f32,
+    ) -> <<Piet as RenderContext>::Text as Text>::TextLayout {
         // TODO: caching of both the format and the layout
-        let format = TextFormat::create(&dwrite_factory)
-            .with_family("Segoe UI")
-            .with_size(15.0)
+        let font = rt
+            .text()
+            .new_font_by_name("Segoe UI", font_size)
+            .unwrap()
             .build()
             .unwrap();
-        let layout = TextLayout::create(&dwrite_factory)
-            .with_text(&self.label)
-            .with_font(&format)
-            .with_width(1e6)
-            .with_height(1e6)
+        // TODO: Layout height and width: 1e6
```

Comment:
This TODO can be removed, though we're going to have to revisit this. The use of 1e6 was basically a hack to get a single line. But down the road we'll probably want a `Label` to be able to do line-breaking to fit the layout width as well. This is dependent on a high-level text interface, which is a glimmer in my eye at the moment.